conftest.py

```diff
@@ -168,6 +195,13 @@ def kill_all_test_spawned_processes() -> None:
         print("Not all processes were killed after test")
 
 
+@pytest.fixture
+def mpi_app_path() -> str:
```

Comment:
How about returning a higher level object like `pathlib.Path` here to simplify the later interactions (e.g. no more os.path.join needed)?

I notice that when used [below](https://github.com/al-rigazzi/SmartSim/pull/7/files#diff-0f1ac619f474c6a82e445241283429b653622770ea569a3b904a655099e7e6f3R75), the test method does `os.path.join(x, "mpi_app")` too.... Can't we return that here?

What's happening on lines 172/174? We have path_to_out = "/something/mpi_app" and then in cmd, we apend `mpi_app` again? Can we avoid adding the same thing twice?

Reply:
It's because I called the executable and its directory with the same name. Let's make a generic app directory in case we need more in the future.

Reply:
OK, used `Path` and tried to refine the logic.